labgrid/driver/usbstoragedriver.py

```diff
@@ -152,6 +160,67 @@ def get_size(self):
         size = subprocess.check_output(self.storage.command_prefix + args)
         return int(size)*512
 
+    @Driver.check_active
+    @step(args=['srcfile'])
+    def write_file(self, srcfile, dstpath="/", partition=None):
+        """
+        Writes the file specified by filename to the filesystem on the bound USB storage
+        root device or partition.
+
+        Args:
+            srcfile(str): path to the file to write to bound USB storage filesystem
+            dstpath (str): optional, destination path to write filename to (defaults to /)
+            partition (int or None): optional, mount the specified partition or None for mounting
+                the root device (defaults to None)
+        """
+        assert srcfile, "write_file requires srcfile"
+        mf = ManagedFile(srcfile, self.storage)
+        mf.sync_to_resource()
+
+        self.await_medium()
+
+        partition = "" if partition is None else partition
+        remote_path = mf.get_remote_path()
+        target = "{}{}".format(self.storage.path, partition)
+
+        label = escape(target)
+        dstpath = os.path.normpath(dstpath)
+        if not dstpath.startswith("/"):
+            raise ValueError("dstpath must be ab absolute path")
+
+        dstpath = "/media/{}/{}".format(label, dstpath)
+
+        self.logger.info('Copying %s to %s via rsync', remote_path, dstpath)
+
+        args = ["pmount {} {}".format(target, label)]
+
+        processwrapper.check_output(
+            self.storage.command_prefix + args,
+            print_on_silent_log=True
+        )
+
+        caught = None
+
+        try:
+            args = ["rsync -v {} {}".format(remote_path, dstpath)]
+
+            processwrapper.check_output(
+                self.storage.command_prefix + args,
+                print_on_silent_log=True
+            )
+        except Exception as e:
+            caught = e
+
+        args = ["pumount {}".format(target)]
+
+        processwrapper.check_output(
+            self.storage.command_prefix + args,
+            print_on_silent_log=True
+        )
+
+        if caught:
+            raise caught
```

Comment:
Please use a more idiomatic try:/finally: structure.

Reply:
I might've never used finally before, so I had forgotten about it. Thanks! :)